### ISSUE TITLE
Overridden private fields trigger runtime error

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/dev_compiler#370]
The following code:

```dart
class A {
  int _x = 42;
}

class B extends A {
  int _x = 95;
}

main() {
  var a = new B();
  print(a._x);
}
```

generates this error:

```
Uncaught TypeError: Cannot convert a Symbol value to a string
virtualField @ _classes.js:289
```

**Comment by jmesserly** [comment]
@vsmenon -- that's an illegal field override with field, right?

**Comment by jmesserly** [comment]
in other words, is this a dupe of #52

**Comment by vsmenon** [comment]
Right, if we statically disallow, we could make this a "won't fix".

**Comment by jmesserly** [comment]
Nice. I'll dupe it as the other bug covers either fix option. I think we're leaning towards restriction first. While runtime fix is pretty easy, it has some bad implications on the generated code.

**Comment by vsmenon** [comment]
I'm hitting a different error if I change the override to a getter:

```dart
class A {
  int _x = 42;
}

class B extends A {
  int get _x => 32;
}

main() {
  var a = new B();
  print(a._x);
}
```

Error is:
```
Uncaught TypeError: Cannot set property Symbol(_x) of [object Object] which has only a getter
_isolate_helper.js:685
```

**Comment by vsmenon** [comment]
We could statically disallow this as well though.

**Comment by jmesserly** [comment]
good catch! I forgot if that was supposed to be supported or not :)

does it work differently with public fields (I guess probably not)?

**Comment by jmesserly** [comment]
Looking at the comments, it looks like we didn't intend to support overriding fields at all, even with getter/setter. (It doesn't interact well with separate compilation. Basically has the same issue.)

I wonder if this works, though:

```
class A {
  int get _x => 42;
}

class B extends A {
  int _x = 32;
}

main() {
  var a = new B();
  print(a._x);
}
```
That one *should* work, but might not, because this code looks pretty fishy:
https://github.com/dart-lang/dev_compiler/blob/ed47365eaa6985199555289616b242d2c2878a43/lib/runtime/_classes.js#L289

FWIW, that concat is meaningless ... the string is only to help debugging.
